File: angular/employee/src/app/shared/data-access/api-error.ts
export interface ApiError {
    /**
     * HTTP Status code (e.g., 409)
     */
    status?: number;

    /**
     * A short, unique error code (e.g., FK_VIOLATION)
     */
    errorCode?: string;

    /**
     * A readable message for the client/user
     */
    message?: string;

    /**
     * Optional: The specific field causing the issue (e.g., departmentId)
     */
    field?: string;
}

File: spring/employee/src/main/java/com/example/employee/models/ApiError.java
package com.example.employee.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ApiError {
	// HTTP Status code (e.g., 409)
    private int status;
    
    // A short, unique error code (e.g., FK_VIOLATION)
    private String errorCode; 
    
    // A readable message for the client/user
    private String message;
    
    // Optional: The specific field causing the issue (e.g., departmentId)
    private String field; 
}
